text/0035-physics.md

```diff
@@ -0,0 +1,175 @@
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Interactions of Advanced Physics Features
+
+### Server Only Simulation
+If all Rigidbody NetworkObjects are controlled by the server (Object is owned by the server or has the NetworkTransform authority set to server) then all Physics features are fully supported.
+- On the server the physics simulation will play out like in a singleplayer games and all interactions triggers, joints, raycasts, collisions, forces etc. will work fine.
+- Clients will receive position updates
+
+### Mixed Simulation Player Collision
+
+If two player controlled dynamic rigidbodies collide this will result in two dynamic/kinematic collisions as if both players where running against an immovable object.
+
+### Mixed Simulation Joints
+
+Joints between two dynamic rigidbodies which are owned by different peers will not resolve properly. This is not a scenario which we can support. The joints will instead act as if all the the Rigidbodies which are not controlled by the controller of the corresponding Rigidbody were kinematic.
+
+### Mixed Simulation Forces
+
+Forces can only be applied to a NetworkRigidbody by the peer which is controlling it. Often forces are applied in a specific area, by applying these forces on every peer all the NetworkRigidbodies will be affected correctly.
+
+### Interaction With Non Networked Physics Objects
+
+NetworkRigidbodies can interact and collide with regular Rigidbodies. Since this is often unwanted behavior it is recommended to either use a different collision layer for non networked rigidbodies or to move them into a separate physics scene entirely.
+
+## Why we do not synchronize velocity
+
+There are two reasons why we would need to synchronize velocity.
+1. Prediction: velocity allows for a more accurate prediction.
+2. Extrapolation in case of packet loss.
+
+We do not support prediction currently so 1. is not relevant. Regarding extrapolation while this might improve the smoothness of the Rigidbody when extrapolating in case of a lag due to packet loss it is non trivial to implement this without resulting in jitter. What we would need to do here is transition the Rigidbody from a simply replicating mode into a dynamic Rigidbody again and run the physics simulation to extrapolate. We currently do not have a good way to make this transition smooth because we are not in control of the FixedUpdate and the physics simulation. Instead simply using the NetworkTransforms extrapolation mode results in a more smooth experience and keeps our implementation simple.
+
+## Why we do not synchronize triggers and collision events
+
+1. Collision and Trigger events contain information in reference types (classes) which is not something we could properly serialize. In addition these events contain information about the components and GameObject involved in the collision. We currently have no way to serialize that information over the network.
```

Comment:
could we have restrictions similar to parenting and make sure we sync at least the collisions with NetworkObjects? We'd be able to sync a net obj's ID.
I'm guessing it'd still be hard to get an ID from a material for example. Wouldn't it still be ok to sync partial information? It'd be better than nothing, no?

Reply:
We might be able to do something but it would be very limited. The collision event contains information such as:
- The collider it collided with (we cannot synchronize this it's not a NetworkBehaviour and one Rigidbody can have many and they can be on child objects.
- Contacts, contact points, velocities, impulse could in theory be synced but they would not reflect the situation on the client at the time the `NetworkCollision` would be fired.

Reply:
we could still have a "DelayedCollisionPoint", telling users this isn't an exact point, but still something that happened in the past if you want to play some sparkle VFX for example?
It's not the mountain I want to die on, if you don't think that'd add value, we can close this :)